hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/RoundRobinVolumeChoosingPolicy.java

```diff
@@ -61,7 +66,7 @@ public HddsVolume chooseVolume(List<HddsVolume> volumes,
     int startVolumeIndex = currentVolumeIndex;
 
     while (true) {
-      final HddsVolume volume = volumes.get(currentVolumeIndex);
+      final HddsVolume volume = volumesWithWriteAllowed.get(currentVolumeIndex);
```

Comment:
`currentVolumeIndex` is calculated based on `volumes.size()`, so this could result in `IndexOutOfBoundsException`.

Volumes should not be eagerly pre-filtered, they can be filtered in the loop.